### PR TITLE
stake-pool-cli: Add best practices for fees, prevent zero fees

### PR DESCRIPTION
**Pull request by joncinque**
#### Problem

People can shoot themselves in the foot by creating pools with no fees.

#### Solution

Only allow creating a pool with zero fees given a `--yolo` flag, and add some best practices around fees.

@bennofs can you take a look specifically at the "Best Practices" section in the Fees page?  I only added absolute minimums there, so we may want to more forcefully suggest higher numbers.  Or you may have calculated some totally different values!

### CODE REVIEW

File: stake-pool/cli/src/main.rs [hunk 1]
```diff
@@ -1928,6 +1958,12 @@ fn main() {
                     .takes_value(true)
                     .help("Stake pool reserve keypair [default: new keypair]"),
             )
+            .arg(
+                Arg::with_name("yolo")
```
Comment (t-nelson) [inline stake-pool/cli/src/main.rs:1962]: Might want to go with a more specific flag name.  You never know when you'll need to yolo something else in the future :wink:

Comment (joncinque) [reply stake-pool/cli/src/main.rs:1962]: hehe, do you suggest something like `--yolo-fees` then?

Comment (bennofs) [reply stake-pool/cli/src/main.rs:1962]: How about `--unsafe-fees`? I'd hope "unsafe" is enough to make people scared?

File: docs/src/stake-pool/fees.md [hunk 2]
```diff
@@ -0,0 +1,145 @@
+---
+title: Fees
+---
+
+The stake pool program provides managers many options for making the pool
+financially viable, predominantly through fees. There are five different sources
+of fees.
```
Comment (bennofs) [inline docs/src/stake-pool/fees.md:7]: I would suggest adding a small disclaimer here that operators of the pool should take time to understand the purpose of the fees  and think about them carefully to ensure that the pool cannot be abused. Right now this looks like a reference that some might just skim quickly.

**Comment by bennofs** [comment]
I the quickstart guide, I think this would be a good place to link to the fee guide as well:

>  In our example, we will use fees of 0.3%, a referral fee of 50%

just to tell people where they can read up on how to decide on fees. 

Other than that, I think just preventing zero fees is okay, as there are IMO no real hard limits on these fees. It depends on how much risk the pool is willing to take, and if it becomes a problem the fees can still be adjusted, so a pool with good monitoring may opt to set lower fees than a "deploy and forget"-style pool :)

File: stake-pool/cli/src/main.rs (hunk 1 above)
Comment (joncinque) [reply stake-pool/cli/src/main.rs:1962]: That probably makes more sense, thanks!

File: docs/src/stake-pool/fees.md (hunk 2 above)
Comment (joncinque) [reply docs/src/stake-pool/fees.md:7]: That's a good point, Will do

**Comment by joncinque** [comment]
Thanks for all the feedback!

**Pull request merged**